lib/services/user_service.dart

```diff
@@ -11,34 +11,37 @@ typedef AppUser = academy.User;
 typedef FirebaseUser = User;
 
 class UserService with ListenableServiceMixin {
-  final _log = getLogger('UserService');
+  final log = getLogger('UserService');
   final _authenticationService = locator<FirebaseAuthenticationService>();
 
   UserService() {
-    listenToReactiveValues([_currentUser]);
-    _authenticationService.firebaseAuth.authStateChanges().listen(_restoreUser);
+    _setUserFromFirebaseAuth(
+      firebaseUser: _authenticationService.firebaseAuth.currentUser,
+    );
   }
 
   AppUser? _currentUser;
   AppUser get currentUser => _currentUser!;
 
   bool get hasUser => _currentUser != null;
 
-  /// Restore User from Firebase Auth on authStateChanges
-  void _restoreUser(FirebaseUser? user) {
-    if (user == null) {
-      _log.i('User is currently signed out!');
-      return;
+  bool get hasFirebaseUser =>
+      _authenticationService.firebaseAuth.currentUser != null;
+
+  /// Sets User from FirebaseAuth
+  void _setUserFromFirebaseAuth({FirebaseUser? firebaseUser}) {
+    if (firebaseUser == null) {
+      _currentUser = null;
+    } else {
+      _currentUser = _extractUserFromFirebaseUser(firebaseUser);
```

Comment:
Interesting. So it's available all the time as the firebase user. This is a good catch. It takes the stream listening out completely. 

Well done spotting this.

Reply:
@FilledStacks 
Exactly, as on Web the Services are rebuilt on refresh or page navigation, there is no need to listen to authStateChanges, we just look for the FirebaseUser on FirebaseAuth, which is more or less as looking into the
backend for the User.

What I'm seeing during this weeks is that if the app is going to be on Web platform apart from mobile and desktop, the app needs to be developed as a dumb client driven by the backend. That force the other platforms to be aligned as the Web and loose some extra functionality or performance done on client side unless we find a way to give Web platform a mechanism under the hood with Stacked to work close to the other platforms. I have an idea but I don't know if it is going to work, we can discuss this later.